tests/database/test_header_db.py

```diff
@@ -142,6 +142,44 @@ def test_can_patch_holes(headerdb, genesis_header):
     assert_headers_eq(headerdb.get_canonical_head(), pseudo_genesis)
 
 
+def test_gap_filling_handles_uncles_correctly(headerdb, genesis_header):
+    headerdb.persist_header(genesis_header)
+    chain_a = mk_header_chain(genesis_header, length=10)
+    chain_b = mk_header_chain(genesis_header, length=10)
+
+    assert headerdb.get_header_chain_gaps() == GENESIS_CHAIN_GAPS
+    # Persist the checkpoint header with a trusted score
+    # chain_a[7] has block number 8 because `chain_a` doesn't include the genesis
+    pseudo_genesis = chain_a[7]
+    headerdb.persist_checkpoint_header(pseudo_genesis, 154618822656)
```

Comment:
Instead of this magic number, maybe `sum(header.difficulty for header in chain_a[0:8])`?